checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1708 @@
+      }
+    }
+
+    // If none of the parameters were @MustCallAlias, it must be the receiver
+    if (result == null && callNode instanceof MethodInvocationNode) {
+      result = ((MethodInvocationNode) callNode).getTarget().getReceiver();
+    }
+
+    result = removeCastsAndGetTmpVarIfPresent(result);
+    return result;
+  }
+
+  /**
+   * If a temporary variable exists for node after typecasts have been removed, return it.
+   * Otherwise, return node.
+   *
+   * @param node a node
+   * @return either a tempvar for node's content sans typecasts, or node
+   */
+  private Node removeCastsAndGetTmpVarIfPresent(Node node) {
+    // TODO create temp vars for TypeCastNodes as well, so we don't need to explicitly remove casts
+    // here
+    node = removeCasts(node);
+    LocalVariableNode tmpVar = typeFactory.getTempVarForNode(node);
+    return tmpVar != null ? tmpVar : node;
+  }
+
+  /**
+   * Get the nodes representing the arguments of a method or constructor invocation from the
+   * invocation node.
+   *
+   * @param node a MethodInvocation or ObjectCreation node
+   * @return a list of the arguments, in order
+   */
+  private List<Node> getArgumentsOfInvocation(Node node) {
+    List<Node> arguments;
+    if (node instanceof MethodInvocationNode) {
+      MethodInvocationNode invocationNode = (MethodInvocationNode) node;
+      arguments = invocationNode.getArguments();
+    } else {
+      if (!(node instanceof ObjectCreationNode)) {
+        throw new BugInCF("unexpected node type " + node.getClass());
+      }
+      arguments = ((ObjectCreationNode) node).getArguments();
+    }
+    return arguments;
+  }
+
+  /**
+   * Get the elements representing the formal parameters of a method or constructor, from an
+   * invocation of that method or constructor.
+   *
+   * @param node a method invocation or object creation node
+   * @return a list of the declarations of the formal parameters of the method or constructor being
+   *     invoked
+   */
+  private List<? extends VariableElement> getFormalsOfInvocation(Node node) {
+    ExecutableElement executableElement;
+    if (node instanceof MethodInvocationNode) {
+      MethodInvocationNode invocationNode = (MethodInvocationNode) node;
+      executableElement = TreeUtils.elementFromUse(invocationNode.getTree());
+    } else {
+      if (!(node instanceof ObjectCreationNode)) {
+        throw new BugInCF("unexpected node type " + node.getClass());
+      }
+      executableElement = TreeUtils.elementFromUse(((ObjectCreationNode) node).getTree());
+    }
+
+    List<? extends VariableElement> formals = executableElement.getParameters();
+    return formals;
+  }
+
+  /**
+   * Does the method being invoked have a not-owning return type?
+   *
+   * @param node a method invocation
+   * @return true iff (1) the checker is not in no-lightweight-ownership mode, (2) the method has a
+   *     non-void return type, and (3) a NotOwning annotation is present on the method declaration
```

Comment:
This method can return true even if the method has a void return type.  (That is, the spec doesn't match the implementation.)